e2e_tests/test_suites/inventoryreporting/inventory_reporting.go

```diff
@@ -0,0 +1,152 @@
+
+// TestSuite is an OSInventoryReporting test suite.
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && !testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range headImageTestSetup() {
+		wg.Add(1)
+		go inventoryReportingTestCase(ctx, setup, tests, &wg, logger, testCaseRegex)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runInventoryReportingTest(ctx context.Context, testSetup *inventoryTestSetup, testCase *junitxml.TestCase, logwg *sync.WaitGroup) {
+	testCase.Logf("Creating compute client")
+
+	computeClient, err := gcpclients.GetComputeClient()
+	if err != nil {
+		testCase.WriteFailure("Error getting compute client: %v", err)
+		return
+	}
+
+	testSetup.hostname = fmt.Sprintf("inventory-test-%s-%s", path.Base(testSetup.testName), utils.RandString(5))
+
+	var metadataItems []*api.MetadataItems
+	metadataItems = append(metadataItems, testSetup.startup)
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("enable-os-inventory", "true"))
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("osconfig-disabled-features", "tasks,guestpolicies"))
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("osconfig-enabled-prerelease-features", "inventoryreporting"))
+
+	testProjectConfig := testconfig.GetProject()
+	zone := testProjectConfig.AcquireZone()
+	defer testProjectConfig.ReleaseZone(zone)
+
+	testCase.Logf("Creating instance %q with image %q", testSetup.hostname, testSetup.image)
+	inst, err := utils.CreateComputeInstance(metadataItems, computeClient, testSetup.machineType, testSetup.image, testSetup.hostname, testProjectConfig.TestProjectID, zone, testProjectConfig.ServiceAccountEmail, testProjectConfig.ServiceAccountScopes)
+	if err != nil {
+		testCase.WriteFailure("Error creating instance: %v", err)
+		return
+	}
+	defer inst.Cleanup()
+	defer inst.RecordSerialOutput(ctx, path.Join(*config.OutDir, testSuiteName), 1)
+
+	testCase.Logf("Waiting for agent install to complete")
+	if _, err := inst.WaitForGuestAttributes("osconfig_tests/install_done", 5*time.Second, 10*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for osconfig agent install: %v", err)
+		return
+	}
+
+	// Build regexes for verification.
+	patterns := []string{
+		`.*Calling ReportInventory with request.*`,
+		fmt.Sprintf(`.*"hostname":.*"%s".*`, testSetup.hostname),
+		fmt.Sprintf(`.*"short_name":.*"%s".*`, testSetup.shortName),
+	}
```

Comment:
Can there also be a pattern check that the call was successful? Or if this does that, can you add a comment? When I see a log message like "Calling ..." I think it's going to happen but hasn't occurred yet.

Reply:
The debug log should also record the response, so we should also just check for that.

Reply:
Just checked and we are not debug-logging the response for report inventory yet, but i can add that in a separate PR. For the time being I added another regex for "Finished task OSInventory" to match the console output with and a negative lookup for "Error reporting inventory" to fail on. Both changes have been tested and should be sufficient until I add in the logging for response and update the tests again once new version is released to staging and unstable.